server/devdocs/bin/generate-components-usage-counts

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env node
+
+/**
+ * This script generates a usage counts for the dependecies of a list of modules.
+ * It accepts a newline-delimited list of .js and|or .jsx files
+ * as its input, and writes the index to server/devdocs/usage-counts.js
+ */
+
+var async = require( 'async' ),
+	fs = require( 'fs' ),
+	fspath = require( 'path' ),
+	root = fspath.dirname( fspath.join( __dirname, '..', '..' ) ),
+
+	// Copyright (c) 2014-present, Facebook, Inc. See CREDITS.md#L1383
+	replacePatterns = {
+		BLOCK_COMMENT_RE: /(?:\/\*(.|\n)*?\*\/)|(\/\/.+(\n|$))/g,
+		LINE_COMMENT_RE: /\/\/.+(\n|$)/g,
+		IMPORT_RE: /(\bimport\s+?(?:.+\s+?from\s+?)?)(['"])([^'"]+)(\2)/g,
+		EXPORT_RE: /(\bexport\s+(?:[^'"]+\s+from\s+)??)(['"])([^'"]+)(\2)/g,
+		REQUIRE_RE: /(\brequire\s*?\(\s*?)(['"])([^'"]+)(\2\s*?\))/g
+	};
+
+function main() {
```

Comment:
I like that you put the `main` function first – made it a bit easier to track where does the execution start.